source/gui/addonStoreGui/viewModels.py

```diff
@@ -0,0 +1,663 @@
+			AvailableAddonStatus.DOWNLOADING: _("Downloading"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.DOWNLOAD_FAILED: _("Download failed"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.DOWNLOAD_SUCCESS: _("Downloaded"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.INSTALLING: _("Installing"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.INSTALL_FAILED: _("Install failed"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.INSTALLED: _("Installed, restart required"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.ENABLED: _("Enabled after restart"),
+			# Translators: Status for addons shown in the add-on store dialog
+			AvailableAddonStatus.RUNNING: _("Enabled"),
+		}
+		return _labels
+
+
+addonStoreStateToAddonHandlerState: Dict[AvailableAddonStatus, addonHandler.AddonStateCategory] = {
+	AvailableAddonStatus.INSTALLED: addonHandler.AddonStateCategory.PENDING_INSTALL,
+}
+
+
+class AddonListItemVM:
+	def __init__(
+			self,
+			model: AddonDetailsModel,
+			status: AvailableAddonStatus = AvailableAddonStatus.AVAILABLE
+	):
+		self._model: AddonDetailsModel = model  # read-only
+		self._status: AvailableAddonStatus = status  # modifications triggers L{updated.notify}
+		self.updated = extensionPoints.Action()  # Notify of changes to VM, argument: addonListItemVM
+
+	@property
+	def model(self):
+		return self._model
+
+	@property
+	def status(self):
+		return self._status
+
+	@status.setter
+	def status(self, newStatus: AvailableAddonStatus):
+		if newStatus != self.status:
+			log.debug(f"addon status change: {self.Id}: status: {newStatus}")
+			self._status = newStatus
+			# ensure calling on the main thread.
+			core.callLater(delay=0, callable=self.updated.notify, addonListItemVM=self)
+
+	@property
+	def Id(self) -> Optional[str]:
+		return self._model.addonId
+
+
+class AddonDetailsVM:
+	def __init__(self, listItem: Optional[AddonListItemVM] = None):
+		self._listItem: Optional[AddonListItemVM] = listItem
+		self.updated = extensionPoints.Action()  # triggered by setting L{self._listItem}
+
+	@property
+	def listItem(self) -> Optional[AddonListItemVM]:
+		return self._listItem
+
+	@listItem.setter
+	def listItem(self, newListItem: Optional[AddonListItemVM]):
+		if (
+			self._listItem == newListItem  # both may be same ref or None
+			or (
+				None not in (newListItem, self._listItem)
+				and self._listItem.Id == newListItem.Id  # confirm with addonId
+			)
+		):
+			# already set, exit early
+			return
+		self._listItem = newListItem
+		# ensure calling on the main thread.
+		core.callLater(delay=0, callable=self.updated.notify, addonDetailsVM=self)
+
+
+class AddonListVM:
+	presentedAttributes = [

```

Comment:
Shouldn't this be a tuple, not list?
```suggestion
	presentedAttributes = (

```